pkg/component/autoscaling/gardenercustommetrics/kubeobjects/deployment.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubeobjects
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+)
+
+func makeDeployment(deploymentName, namespace, containerImageName, serverSecretName string) *appsv1.Deployment {
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deploymentName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				v1beta1constants.LabelApp: gcmxBaseName,
+				// The actual availability requirement of gardener-custom-metrics is closer to the "controller"
+				// availability level (even less, actually). The value below is set to "server" solely to satisfy
+				// the requirement for consistency with existing components.
+				resourcesv1alpha1.HighAvailabilityConfigType: "server",
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas:             ptr.To[int32](1),
+			RevisionHistoryLimit: ptr.To[int32](2),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					v1beta1constants.LabelApp:   gcmxBaseName,
+					v1beta1constants.GardenRole: gcmxBaseName,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						v1beta1constants.LabelApp:                                                  gcmxBaseName,
+						v1beta1constants.GardenRole:                                                gcmxBaseName,
+						v1beta1constants.LabelNetworkPolicyToDNS:                                   "allowed",
```

Comment:
```suggestion
						v1beta1constants.LabelNetworkPolicyToDNS:                                   v1beta1constants.LabelNetworkPolicyAllowed,
```
apply this to other occurrences

Reply:
Done. Thanks for the pointer!